### ISSUE TITLE
 ERROR o.x.i.i.DefaultTasksManager on the logs during confluence migration

### ISSUE THREAD

**Issue by mouhb** [xwikisas/application-confluence-migrator-pro#51]
Confluence Migrator version : 1.7.9
XWiki version : 14.10.17

The following error is displayed on tomcat logs during Confluence migrations :

```
2024-01-15 14:40:49,453 [task-manager-consumer] ERROR o.x.i.i.DefaultTasksManager    - [org.xwiki.index.internal.TaskData@5917845b[timestamp=1705326049429,docId=8205558590638710401,type=mention,attempts=10,stop=false,wikiId=xwiki,version=71.2]] abandoned because it has failed too many times. 
org.xwiki.index.IndexException: Failed during the mention task execution
	at org.xwiki.mentions.internal.DefaultMentionsDataConsumer.consume(DefaultMentionsDataConsumer.java:104)
	at org.xwiki.index.internal.TaskExecutor.internalExecute(TaskExecutor.java:103)
	at org.xwiki.index.internal.TaskExecutor.execute(TaskExecutor.java:83)
	at org.xwiki.index.internal.DefaultTasksManager.consume(DefaultTasksManager.java:226)
	at org.xwiki.index.internal.DefaultTasksManager.run(DefaultTasksManager.java:209)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: com.xpn.xwiki.XWikiException: Error number 3103 in 3: Exception while reading version [71.1] for document id [8,205,558,590,638,710,401]
	at com.xpn.xwiki.doc.XWikiDocumentArchive.loadDocument(XWikiDocumentArchive.java:404)
	at com.xpn.xwiki.store.XWikiHibernateVersioningStore.loadXWikiDoc(XWikiHibernateVersioningStore.java:213)
	at com.xpn.xwiki.internal.doc.DatabaseDocumentRevisionProvider.loadRevision(DatabaseDocumentRevisionProvider.java:95)
	at com.xpn.xwiki.internal.doc.DatabaseDocumentRevisionProvider.getRevision(DatabaseDocumentRevisionProvider.java:78)
	at com.xpn.xwiki.internal.doc.DatabaseDocumentRevisionProvider.getRevision(DatabaseDocumentRevisionProvider.java:61)
	at com.xpn.xwiki.internal.doc.DefaultDocumentRevisionProvider.getRevision(DefaultDocumentRevisionProvider.java:103)
	at org.xwiki.mentions.internal.DefaultMentionsDataConsumer.consume(DefaultMentionsDataConsumer.java:96)
	... 5 common frames omitted
Caused by: com.xpn.xwiki.XWikiException: Error number 2002 in 2: Error parsing xml
	at com.xpn.xwiki.doc.XWikiDocument.fromXML(XWikiDocument.java:5317)
	at com.xpn.xwiki.doc.XWikiDocument.fromXML(XWikiDocument.java:5328)
	at com.xpn.xwiki.doc.XWikiDocument.fromXML(XWikiDocument.java:5292)
	at com.xpn.xwiki.doc.XWikiDocumentArchive.loadDocument(XWikiDocumentArchive.java:390)
	... 11 common frames omitted
Caused by: org.xwiki.filter.FilterException: Failed to read XAR XML document
	at org.xwiki.filter.xar.internal.input.XARInputFilterStream.readDocument(XARInputFilterStream.java:123)
	at org.xwiki.filter.xar.internal.input.XARInputFilterStream.read(XARInputFilterStream.java:71)
	at org.xwiki.filter.xar.internal.input.XARInputFilterStream.read(XARInputFilterStream.java:47)
	at org.xwiki.filter.input.AbstractBeanInputFilterStream.read(AbstractBeanInputFilterStream.java:79)
	at com.xpn.xwiki.internal.filter.XWikiDocumentFilterUtils.importEntity(XWikiDocumentFilterUtils.java:191)
	at com.xpn.xwiki.doc.XWikiDocument.fromXML(XWikiDocument.java:5314)
	... 14 common frames omitted
Caused by: com.ctc.wstx.exc.WstxUnexpectedCharException: Unexpected character 'd' (code 100) in prolog; expected '<'
 at [row,col {unknown-source}]: [1,1]
	at com.ctc.wstx.sr.StreamScanner.throwUnexpectedChar(StreamScanner.java:666)
	at com.ctc.wstx.sr.BasicStreamReader.nextFromProlog(BasicStreamReader.java:2148)
	at com.ctc.wstx.sr.BasicStreamReader.next(BasicStreamReader.java:1180)
	at com.ctc.wstx.sr.BasicStreamReader.nextTag(BasicStreamReader.java:1203)
	at org.xwiki.filter.xar.internal.input.DocumentLocaleReader.read(DocumentLocaleReader.java:358)
	at org.xwiki.filter.xar.internal.input.DocumentLocaleReader.read(DocumentLocaleReader.java:335)
	at org.xwiki.filter.xar.internal.input.XARInputFilterStream.readDocument(XARInputFilterStream.java:121)
	... 19 common frames omitted
```

The error is preceded with lots of these warnings :

```
at [row,col {unknown-source}]: [1,1]]. 
2024-01-15 14:40:49,203 [task-manager-consumer] WARN  o.x.i.i.DefaultTasksManager    - Error during the execution of task [org.xwiki.index.internal.TaskData@247e03bc[timestamp=1705326049202,docId=8205558590638710401,type=mention,attempts=3,stop=false,wikiId=xwiki,version=71.2]] for document [xwiki:ITDOCS.Tapes.LTO8 Tapes.WebHome]. Cause: [WstxUnexpectedCharException: Unexpected character 'd' (code 100) in prolog; expected '<'
 at [row,col {unknown-source}]: [1,1]]. 
2024-01-15 14:40:49,241 [task-manager-consumer] WARN  o.x.i.i.DefaultTasksManager    - Error during the execution of task [org.xwiki.index.internal.TaskData@247e03bc[timestamp=1705326049221,docId=8205558590638710401,type=mention,attempts=4,stop=false,wikiId=xwiki,version=71.2]] for document [xwiki:ITDOCS.Tapes.LTO8 Tapes.WebHome]. Cause: [WstxUnexpectedCharException: Unexpected character 'd' (code 100) in prolog; expected '<'
 at [row,col {unknown-source}]: [1,1]]. 
2024-01-15 14:40:49,243 [task-manager-consumer] WARN  o.x.i.i.DefaultTasksManager    - Error during the execution of task [org.xwiki.index.internal.TaskData@5917845b[timestamp=1705326049241,docId=8205558590638710401,type=mention,attempts=1,stop=false,wikiId=xwiki,version=71.2]] for document [xwiki:ITDOCS.Tapes.LTO8 Tapes.WebHome]. Cause: [WstxUnexpectedCharException: Unexpected character 'd' (code 100) in prolog; expected '<'
 at [row,col {unknown-source}]: [1,1]]. 
2024-01-15 14:40:49,270 [task-manager-consumer] WARN  o.x.i.i.DefaultTasksManager    - Error during the execution of task [org.xwiki.index.internal.TaskData@5917845b[timestamp=1705326049261,docId=8205558590638710401,type=mention,attempts=2,stop=false,wikiId=xwiki,version=71.2]] for document [xwiki:ITDOCS.Tapes.LTO8 Tapes.WebHome]. Cause: [WstxUnexpectedCharException: Unexpected character 'd' (code 100) in prolog; expected '<'
```

**Comment by trrenty** [comment]
I believe that this issue was caused by https://jira.xwiki.org/browse/XWIKI-21801 and subsequently fixed by this PR #55. 

Some versions of the imported pages were corrupted due to that bug and could not be processed in any way. Since the mentions were to be retrieved from the content of those corrupted versions, these logs would appear. Similar logs also appear with the Application Task Manager Pro when task macros are located on the homepage of the imported space, if I remember correctly.

However, I'm not 100% sure and it should be tested. Can you try to reproduce the issue with version 1.8.3 of the application, @mouhb ?

**Comment by mouhb** [comment]
> I believe that this issue was caused by https://jira.xwiki.org/browse/XWIKI-21801 and subsequently fixed by this PR #55.
> 
> Some versions of the imported pages were corrupted due to that bug and could not be processed in any way. Since the mentions were to be retrieved from the content of those corrupted versions, these logs would appear. Similar logs also appear with the Application Task Manager Pro when task macros are located on the homepage of the imported space, if I remember correctly.
> 
> However, I'm not 100% sure and it should be tested. Can you try to reproduce the issue with version 1.8.3 of the application, @mouhb ?

I will test it ASAP and then comment the ticket.

**Comment by trrenty** [comment]
Closing the issue as it was most probably fixed by #55. To be reopened if it can be reproduced again.
